src/libcore/slice/mod.rs

```diff
@@ -1706,13 +1715,14 @@ impl<T> [T] {
     ///         v == [2, 1, -3, 4, -5] ||
     ///         v == [2, 1, -3, -5, 4]);
     /// ```
```

Comment:
Docs should mention when this panics. Describe conditions on index. Somehow it should be clear that `index < len` strictly and thus empty slices can't be partitioned.

Reply:
Done.